pkg/controllers/disruption/suite_test.go

```diff
@@ -1361,6 +1366,358 @@ var _ = Describe("Candidate Filtering", func() {
+
+		ExpectMetricCounterValue("karpenter_disruption_actions_performed_total", 1, map[string]string{
+			"action":             "delete",
+			"method":             "consolidation",
+			"consolidation_type": "multi",
+		})
+		ExpectMetricCounterValue("karpenter_disruption_nodes_disrupted_total", 2, map[string]string{
+			"action":             "delete",
+			"method":             "consolidation",
+			"consolidation_type": "multi",
+		})
+		ExpectMetricCounterValue("karpenter_disruption_pods_disrupted_total", 2, map[string]string{
+			"action":             "delete",
+			"method":             "consolidation",
+			"consolidation_type": "multi",
+		})
+	})
+	It("should fire metrics for multi-node replace disruption", func() {
+		nodeClaims, nodes := test.NodeClaimsAndNodes(3, v1beta1.NodeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					v1beta1.NodePoolLabelKey:     nodePool.Name,
+					v1.LabelInstanceTypeStable:   mostExpensiveInstance.Name,
+					v1beta1.CapacityTypeLabelKey: mostExpensiveOffering.CapacityType,
+					v1.LabelTopologyZone:         mostExpensiveOffering.Zone,
+				},
+			},
+			Status: v1beta1.NodeClaimStatus{
+				Allocatable: map[v1.ResourceName]resource.Quantity{
+					v1.ResourceCPU:  resource.MustParse("32"),
+					v1.ResourcePods: resource.MustParse("100"),
+				},
+			},
+		})
+
+		// create our RS so we can link a pod to it
+		rs := test.ReplicaSet()
+		ExpectApplied(ctx, env.Client, rs)
+		pods := test.Pods(4, test.PodOptions{
+			ObjectMeta: metav1.ObjectMeta{Labels: labels,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion:         "apps/v1",
+						Kind:               "ReplicaSet",
+						Name:               rs.Name,
+						UID:                rs.UID,
+						Controller:         ptr.Bool(true),
+						BlockOwnerDeletion: ptr.Bool(true),
+					},
+				},
+			},
+		})
+
+		ExpectApplied(ctx, env.Client, rs, pods[0], pods[1], pods[2], pods[3], nodeClaims[0], nodes[0], nodeClaims[1], nodes[1], nodeClaims[2], nodes[2], nodePool)
+
+		// bind pods to nodes
+		ExpectManualBinding(ctx, env.Client, pods[0], nodes[0])
+		ExpectManualBinding(ctx, env.Client, pods[1], nodes[1])
+		ExpectManualBinding(ctx, env.Client, pods[2], nodes[2])
+		ExpectManualBinding(ctx, env.Client, pods[3], nodes[2])
+
+		// inform cluster state about nodes and nodeclaims
+		ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*v1.Node{nodes[0], nodes[1], nodes[2]}, []*v1beta1.NodeClaim{nodeClaims[0], nodeClaims[1], nodeClaims[2]})
+
+		fakeClock.Step(10 * time.Minute)
+
+		var wg sync.WaitGroup
+		ExpectTriggerVerifyAction(&wg)
+		ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKey{})
+		wg.Wait()
+
+		ExpectMetricCounterValue("karpenter_disruption_actions_performed_total", 1, map[string]string{
+			"action":             "replace",
+			"method":             "consolidation",
+			"consolidation_type": "multi",
+		})
```

Comment:
nit: can we use the proper constants for these and other labels + value?

Reply:
I tend to avoid using constants in testing. The reason: it can mask weird mistakes where you mess with your constant or variable and you didn't intend to. If you test against a know value that you always expect to be there, then you can be assured that if you constant accidentally changes, you will catch it in your testing.

Reply:
fair principle